org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/ObjectComparision.java

```diff
@@ -0,0 +1,654 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.debug.ui;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+
+import org.eclipse.debug.core.DebugException;
+import org.eclipse.debug.core.model.IVariable;
+import org.eclipse.jdt.debug.core.IJavaArray;
+import org.eclipse.jdt.debug.core.IJavaFieldVariable;
+import org.eclipse.jdt.debug.core.IJavaObject;
+import org.eclipse.jdt.debug.core.IJavaValue;
+import org.eclipse.jdt.debug.core.IJavaVariable;
+import org.eclipse.jdt.internal.debug.core.model.JDIArrayValue;
+import org.eclipse.jdt.internal.debug.core.model.JDINullValue;
+
+public class ObjectComparision {
+
+	public static final String IMMEDIATE_RESULT_KEY = "ImmediateResult"; //$NON-NLS-1$
+	public static final String RESULT_FAILED = "ResultFailed"; //$NON-NLS-1$
+
+	public static final String SELECTED_OBJECT_1 = "Selection1"; //$NON-NLS-1$
+	public static final String SELECTED_OBJECT_2 = "Selection2"; //$NON-NLS-1$
+
+	public static final String KEYSET_1 = "keySet1"; //$NON-NLS-1$
+	public static final String KEYSET_2 = "keySet2"; //$NON-NLS-1$
+
+	public static final String VALUESET_1 = "valueSet1"; //$NON-NLS-1$
+	public static final String VALUESET_2 = "valueSet2"; //$NON-NLS-1$
+
+	/**
+	 * Extracts the actual value of the given object
+	 *
+	 * @param value
+	 *            IJavaValue of the selected object
+	 * @throws DebugException
+	 * @return returns actual value of the object or return combination object's name and id if extraction is not possible
+	 */
+	@SuppressWarnings("nls")
+	private String objectExtraction(IJavaValue value) throws DebugException {
+		if (value.getReferenceTypeName().equals("java.lang.String")) {
+			return value.toString();
+		}
+		if (value.getReferenceTypeName().startsWith("java.lang.StringBu")) {
```

Comment:
java.lang.StringBuffer or StringBuilder

Reply:
for both actually

Reply:
then please verify both. Maybe JDK later offers java.lang.StringBungeejumper, that should not break your code.

Reply:
verified both now